src/utils/c_string_util.c

```diff
@@ -19,6 +19,13 @@ char *strncpy_or_abort(char *dest, const char *src, size_t n) {
     return dest;
 }
 
+char *strtoupper(char *s) {
+    for(size_t i = 0; s[i] != '\0'; ++i) {
+        s[i] = toupper(s[i]);
```

Comment:
```suggestion
        s[i] = toupper((unsigned char)s[i]);
```
passing negative numbers to toupper is UB.
From cppreference.com:
> int ch - character to be converted. If the value of ch is not representable as unsigned char and does not equal EOF, the behavior is undefined.

Reply:
Dropped strtoupper()